### PR TITLE
A number of small tweaks to the MVP.

### PR DESCRIPTION
**Pull request by mccheah**
- Master protocol defaults to https if not specified
- Removed upload driver extra classpath functionality
- Added ability to specify main app resource with container:// URI
- Shut down the REST server and thus the driver pod once the application completes
- Updated docs to reflect all of the above
- Add examples to Docker images, mostly for integration testing but could be useful for easily getting started without shipping anything to the cluster

### CODE REVIEW

**Comment by mccheah** [comment]
@foxish @ash211 @erikerlandson bunch of small things here, though in retrospect each of these should be its own pull request.

File: docs/running-on-kubernetes.md [hunk 1]
```diff
@@ -42,11 +42,10 @@ are set up as described above:
       --conf spark.kubernetes.executor.docker.image=registry-host:5000/spark-executor:latest
       examples/jars/spark_2.11-2.2.0.jar
 
-<!-- TODO master should default to https if no scheme is specified -->
 The Spark master, specified either via passing the `--master` command line argument to `spark-submit` or by setting
 `spark.master` in the application's configuration, must be a URL with the format `k8s://<api_server_url>`. Prefixing the
 master string with `k8s://` will cause the Spark application to launch on the Kubernetes cluster, with the API server
-being contacted at `api_server_url`. The HTTP protocol must also be specified.
+being contacted at `api_server_url`. If no HTTP protocol is specified in the URL, it defaults to `https`.
```
Comment (ash211) [inline docs/running-on-kubernetes.md]: Provide an example of what this looks like.  The double protocol thing is confusing as-is

Comment (mccheah) [reply docs/running-on-kubernetes.md]: Done.

File: resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/rest/kubernetes/KubernetesSparkRestServer.scala [hunk 2]
```diff
@@ -237,7 +263,8 @@ private[spark] object KubernetesSparkRestServer {
       parsedArguments.host.get,
       parsedArguments.port.get,
       sparkConf,
-      secretBytes)
+      secretBytes,
+      barrier)
```
Comment (foxish) [inline resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/rest/kubernetes/KubernetesSparkRestServer.scala:267]: Why was `barrier` added?

Comment (mccheah) [reply resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/rest/kubernetes/KubernetesSparkRestServer.scala:267]: The barrier needs to count down to unblock the main thread whenever we stop the server.

**Comment by ash211** [comment]
@mccheah can you please handle merge conflicts?

@foxish anything else you'd want to see changed here before merging?

**Comment by foxish** [comment]
I still haven't gone over this change completely as I do not completely understand some parts of it. I'd like to go over it once tonight. Can we merge this tomorrow?

**Comment by foxish** [comment]
LGTM

**Comment by mccheah** [comment]
@foxish @ash211 fixed merge conflicts. Integration tests were also failing from the pod and service name change, so fixed those also.

**Review by ash211 — APPROVED** [review]
@foxish @erikerlandson either of you have any concerns?  Otherwise let's merge this

**Comment by erikerlandson** [comment]
LGTM

**Pull request merged**